test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,255 @@
+import 'dart:io';
+import 'dart:math';
+
+import 'package:dio/dio.dart';
+import 'package:http_mock_adapter/http_mock_adapter.dart';
+import 'package:requests_signature_dart/src/client/requests_signature_options.dart';
+import 'package:test/test.dart';
+import 'package:requests_signature_dart/requests_signature_dart.dart';
+
+void main() {
+  group('Requests Signature Interceptor Test', () {
+    final testClientId = 'test_client_id';
+    final testClientSecret = 'test_client_secret';
+    final headerName = 'X-Signature';
+    final signaturePattern = '{ClientId}:{Nonce}:{Timestamp}:{SignatureBody}';
+    final baseEndpoint = 'https://google.ca/';
+    final emptyDataInjection = '{}'; //Injecting empty data as String
+
+    late Dio dio;
+    late DateTime now;
+
+    setUp(() {
+      dio = Dio(BaseOptions(
+        validateStatus: (status) => true,
+      ));
+      now = DateTime.now().toUtc();
+    });
+
+    tearDown(() {
+      dio.close();
+    });
+
+    test('Interceptor adds signature header to GET request.', () async {
+      // Arrange
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+      );
+
+      final interceptor = RequestsSignatureInterceptor(options, dio);
+
+      dio.interceptors.add(interceptor);
+
+      // Act
+      final response = await dio.get(baseEndpoint);
+
+      // Assert
+      expect(response.statusCode, 200);
+
+      expect(response.requestOptions.headers, contains(headerName));
+    });
+
+    var httpStatusCoveredByAutoRetry = {
+      HttpStatus.unauthorized,
+      HttpStatus.forbidden,
+    };
+
+    httpStatusCoveredByAutoRetry.forEach(
+      (serverHttpStatus) {
+        test(
+          'Interceptor retries when receiving $serverHttpStatus and auto-retry is enabled and the tolerance is high enough.',
+          () async {
+            // Arrange
+            final clockskewMS = Random().nextInt(1000) + 10;
```

Comment:
Ideally, I wanted to replicate `AutoFixture` where it would use a different value every time for the clockskewMS, to ensure the test works no matter the value set.

Reply:
I've rework that logic and added comments